### PR TITLE
[chore] use slices.Concat instead of a series of append

### PR DESCRIPTION
**Pull request by codeboten**
Follows https://github.com/open-telemetry/opentelemetry-collector/pull/10869

### CODE REVIEW

File: .github/workflows/api-compatibility.yml [hunk 1]
```diff
@@ -35,7 +34,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
         with:
-          go-version: ~1.21.13
+          go-version: ~1.22.6
```
Comment (atoulme) [inline .github/workflows/api-compatibility.yml:37]: this might be worth its own PR?

Comment (codeboten) [reply .github/workflows/api-compatibility.yml]: the PR was built on top of the update to go 1.22, it's ready to review now

**Review by bogdandrutu — APPROVED** [review]

**Review by atoulme — APPROVED** [review]
